src/snowflake/connector/s3_storage_client.py

```diff
@@ -352,7 +356,7 @@ def get_file_header(self, filename: str) -> Optional[FileHeader]:
             None if HEAD returns 404, otherwise a FileHeader instance populated
             with metadata
         """
-        path = quote(self.s3location.path + filename.lstrip("/"))
+        path = self._url_quote(self.s3location.path + filename.lstrip("/"))
```

Comment:
I think it'd be better to not call a static function through `self`.  It's more explicit that this is a static method
```suggestion
        path = SnowflakeS3RestClient._url_quote(self.s3location.path + filename.lstrip("/"))
```